invenio_communities/models.py

```diff
@@ -311,11 +316,19 @@ def remove_record(self, record):
         :type record: `invenio_records.api.Record`
         """
         key = current_app.config['COMMUNITIES_RECORD_KEY']
+        access_key = '_access'
 
         assert self.id in record.get(key, [])
-
         record[key] = [c for c in record[key] if c != self.id]
 
+        assert self.id in record.get(access_key, {'read': []})['read']
+        record[access_key]['read'] = \
+            [c for c in record[access_key]['read'] if c != self.id]
```

Comment:
I dont know if it's worth to care about the complexity for list that we expect to be relatively short.
Anyway i always try to keep the style of the pre-existing code when there are no big issue about that.